commit efbae3505b2a8817e8291468247c5b1f20823a5f (refs/changes/73/3973/1)
Author: Mike Zeller <mike.zeller@joyent.com>
Date:   2018-05-18T08:10:34-07:00 (1 year, 5 months ago)
    
    TRITON-394 cleanupNics can be skipped if the workflow fails early enough

diff --git a/lib/workflows/job-common.js b/lib/workflows/job-common.js
index 8ae4dea..c07aefa 100644
--- a/lib/workflows/job-common.js
+++ b/lib/workflows/job-common.js
@@ -970,7 +970,14 @@ function cleanupNics(job, cb) {
     var macs = job.params.macs;
 
     if (!macs) {
-            var nics = job.params['add_nics'] || job.params['nics'];
+            /*
+             * filteredNetworks.nics will contain any pre provisioned NICs. If
+             * the workflow fails early enough the other job.params fields will
+             * not yet have been populated yet, but we still have some NICs that
+             * need to be removed.
+             */
+            var nics = job.params['add_nics'] || job.params['nics'] ||
+                job.params.filteredNetworks.nics;
 
             if (!nics) {
                 return cb(null, 'No MACs given, and no NICs were provisioned');
